### PR TITLE
Feature/sumologic connection

### PR DESCRIPTION
**Pull request by anandkumarpatel**
* Added `sumologic_connection` resource. 
* Added docs for `sumologic_connection`
I only implemented `WebhookConnection` for now but it can be easily expanded.

### CODE REVIEW

File: sumologic/resource_sumologic_connection.go [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+				},
+				Optional: true,
+				ForceNew: false,
+				Default:  map[string]interface{}{},
+			},
+			"default_payload": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     false,
+				ValidateFunc: validation.StringIsJSON,
```
Comment (seansain) [inline sumologic/resource_sumologic_connection.go]: Might want to add a diffSupressFunc to not throw any errors on JSON differences.
```
DiffSuppressFunc: func(k, old, new string, d *schema.ResourceData) bool {
					newJson, _ := structure.NormalizeJsonString(new)
					oldJson, _ := structure.NormalizeJsonString(old)
					return newJson == oldJson
				},
```

File: sumologic/resource_sumologic_connection_test.go [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+		)
+		return f(s)
+	}
+}
+
+func testAccCheckConnectionDestroy(connection Connection) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		client := testAccProvider.Meta().(*Client)
+		_, err := client.GetConnection(connection.ID)
+		if err == nil {
```
Comment (seansain) [inline sumologic/resource_sumologic_connection_test.go]: In the client we return `nil, nil` when the connection does not exist. So we should check that both the object and the err are `nil`. Right now it always complains that the object is never destroyed even though it is actually being destroyed.

File: sumologic/resource_sumologic_connection_test.go [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckConnectionDestroy(connection),
+		Steps: []resource.TestStep{
+			{
+				Config: createConnectionConfig(name, connectionType, description, url, webhookType, defaultPayload),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckConnectionExists("sumologic_connection.test", &connection, t),
+					testAccCheckConnectionAttributes("sumologic_connection.test"),
+					resource.TestCheckResourceAttr("sumologic_connection.test", "connectionType", connectionType),
```
[the remark this answers is not in the document]
Comment (anandkumarpatel) [reply sumologic/resource_sumologic_connection_test.go:33]: nice catch! fixed

File: sumologic/resource_sumologic_connection_test.go [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+					resource.TestCheckResourceAttr("sumologic_connection.test", "name", name),
+					resource.TestCheckResourceAttr("sumologic_connection.test", "description", description),
+					resource.TestCheckResourceAttr("sumologic_connection.test", "url", url),
+					resource.TestCheckResourceAttr("sumologic_connection.test", "defaultPayload", defaultPayload),
```
[the remark this answers is not in the document]
Comment (anandkumarpatel) [reply sumologic/resource_sumologic_connection_test.go:37]: nice catch!  Do you have an example output of this? I don't have access to a dev sumologic I can run these integration test on.

File: sumologic/resource_sumologic_connection_test.go [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+					resource.TestCheckResourceAttr("sumologic_connection.test", "webhookType", webhookType),
```
[the remark this answers is not in the document]
Comment (anandkumarpatel) [reply sumologic/resource_sumologic_connection_test.go:38]: nice catch! fixed

File: sumologic/resource_sumologic_connection_test.go [hunk 6]
```diff
@@ -0,0 +1,144 @@
...
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/terraform"
+)
+
+//Testing create functionality for Connection resources
+func TestAccConnection_create(t *testing.T) {
+	connectionType := "WebhookConnection"
+	name := acctest.RandomWithPrefix("tf-connection-test-name")
+	description := acctest.RandomWithPrefix("tf-connection-test-description")
+	url := acctest.RandomWithPrefix("https://")
```
[the remark this answers is not in the document]
Comment (anandkumarpatel) [reply sumologic/resource_sumologic_connection_test.go:17]: Ahh ok, will hardcode an example URL then.

File: sumologic/resource_sumologic_connection.go (hunk 1 above)
Comment (anandkumarpatel) [reply sumologic/resource_sumologic_connection.go:75]: Good Idea

File: sumologic/resource_sumologic_connection_test.go (hunk 2 above)
Comment (anandkumarpatel) [reply sumologic/resource_sumologic_connection_test.go:124]: fixed

**Comment by seansain** [comment]
Wasn't sure how to push to your branch, so I created a new PR: #148 
This way we can run all our travis checks.

**Comment by anandkumarpatel** [comment]
moved to PR: https://github.com/SumoLogic/sumologic-terraform-provider/pull/148

**Pull request closed without merging**